### PR TITLE
Add better error message when using @function in class

### PR DESCRIPTION
**Pull request by saulshanabrook**
Addresses https://github.com/egraphs-good/egglog-python/issues/229 cc @shinawy hopefully this would be more helpful?

### CODE REVIEW

File: python/egglog/egraph.py [hunk 1]
```diff
@@ -577,23 +577,25 @@ def _generate_class_decls(  # noqa: C901,PLR0912
...
+                merge,
+                on_merge,
+                mutates,
+                builtin,
+                ruleset=ruleset,
+                unextractable=unextractable,
+                subsume=subsume,
+            )
+        except ValueError as e:
+            raise ValueError(f"Error processing {cls_name}.{method_name}: {e}") from e
```
Comment (hameerabbasi) [inline python/egglog/egraph.py:598]:
```suggestion
            raise ValueError(f"Error processing {cls_name}.{method_name}") from e
```

The `from e` should already give the previous stack trace.

Comment (hameerabbasi) [reply python/egglog/egraph.py:598]: Or explicitly pull out the message.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Minor suggestion.

**Comment by shinawy** [comment]
@saulshanabrook Thank you, it is better now.

File: python/egglog/egraph.py (hunk 1 above)
Comment (saulshanabrook) [reply python/egglog/egraph.py:598]: Thanks for the suggestion! I decided to keep it for now to make it easier so you don't have to find the other traceback as well.

I think just taking the string of the message does an OK job of pulling it out?

**Pull request merged**